engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -133,15 +139,16 @@ private void logIntermediateRendererListForDebugging(List<Node> orderedNodes) {
                 }
 
                 // Reset all the persistentStateChanges that are not requestedStateChanges.
-                Iterator<Map.Entry<Class<?>, StateChange>> iterator = persistentStateChanges.entrySet().iterator();
+                iterator = persistentStateChanges.entrySet().iterator();
                 while (iterator.hasNext()) {
-                    Map.Entry<Class<?>, StateChange> entry = iterator.next();
+                    entry = iterator.next();
                     Class<?> key = entry.getKey();
+                    StateChange stateChangeAgainstKey = entry.getValue();
```

Comment:
Ok, I think we are getting to a compromise here. Please replace the previous two lines with:

```
key = entry.getKey();
stateChange = entry.getValue();
```

by moving the variable declarations at the beginning of the method, as you did with other variables, and I'll be ok with it.